### PR TITLE
Cleanup `test/rocmlir-driver/auto_e2e`

### PR DESCRIPTION
**Pull request by zhanglx13**
This PR converts leftover tests in `auto_e2e` into toml configurations, therefore, the `auto_e2e` dir is deleted.
- Tests in `xdlops`, i.e. `conv2d_f16_wrw_perf_config.mlir` and `conv2d_f32_wrw_perf_config.mlir` are converted into `conv2d_wrw_perf_config.toml`
- `conv2d_wrw_perf_config.cfg` is used to make sure tests in `conv2d_wrw_perf_config.toml` are ran only on mfma-supported GPUs
- The test in `conv2d_f16_wrw_kblocks.mlir` is added into `conv2d_regression_bwd.toml`

### CODE REVIEW

File: mlir/test/e2e/CMakeLists.txt
Comment (krzysz00) [inline mlir/test/e2e/CMakeLists.txt]: I'd put the `${lit_cfg_file}` in quotes in case someone has spaces in their `/path/to/rocmlir`

File: mlir/test/e2e/MixedConvLayouts.toml
Comment (krzysz00) [inline mlir/test/e2e/MixedConvLayouts.toml]: Nit: Missing newline.

File: mlir/test/e2e/conv2d_wrw_perf_config.toml
Comment (krzysz00) [inline mlir/test/e2e/conv2d_wrw_perf_config.toml]: Nit: Newline

File: mlir/test/e2e/conv2d_wrw_perf_config.cfg
Comment (krzysz00) [inline mlir/test/e2e/conv2d_wrw_perf_config.cfg]: Is `config.unsupported` inherited from parent directories? That is, do you need those first two terms in this `if`?

**Review by krzysz00 — APPROVED** [review]
Nice one, I'm approving with the understanding that my nits get fixed before you hit merge.

File: mlir/test/e2e/conv2d_wrw_perf_config.cfg
Comment (krzysz00) [inline mlir/test/e2e/conv2d_wrw_perf_config.cfg]: Nit: missing final newline (if you're in VS code, there might be a setting you'll need to enable to get this automatically, IIRC)

**Review by krzysz00 — APPROVED** [review]
LGTM, and I take back the comment about `cp` earlier

File: mlir/test/e2e/conv2d_wrw_perf_config.cfg
Comment (zhanglx13) [reply mlir/test/e2e/conv2d_wrw_perf_config.cfg]: Why do we want a newline? Is it coding guide or something?

Comment (krzysz00) [reply mlir/test/e2e/conv2d_wrw_perf_config.cfg]: Yeah, files ending in a `\n` is a pretty standard bit of coding style, to the point most editors are or can be made to enforce it

The main reason is that, without one, `cat`ing your file will mess up the shell prompt by leaving it on the same line as the end of the file

But also, not having `\n` on the last line can confuse various scripts

**Pull request merged**
